Sprint-2/stretch/count-words.js

```diff
@@ -19,10 +17,27 @@
     - Setting values on an object
 
 ## Advanced challenges
-
 1. Remove all of the punctuation (e.g. ".", ",", "!", "?") to tidy up the results
-
 2. Ignore the case of the words to find more unique words. e.g. (A === a, Hello === hello)
-
 3. Order the results to find out which word is the most common in the input
 */
+
+function countWords(str) {
+
+  counter = {}
+  str = str.toLowerCase().split(' ');
+
+  for (let word of str) {
+    word = word.replace(/[^a-z]/g, "");
+
+    if (word in counter) {
+      counter[word] += 1
+    } else {
+      counter[word] = 1;
+    }
+  }
+
+  return counter
+}
+
```

Comment:
What do you expect from the following function call?
Does your function return the value you expected?

```
countWords("hello.world hello world hello$world");
```

Reply:
the function returned what expected by how I understood  the task description:
{
  "helloworld": 2,
  "hello": 1,
  "world": 1
}